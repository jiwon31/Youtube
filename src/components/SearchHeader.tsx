import { useEffect, useState } from "react";
import { AiFillYoutube, AiOutlineSearch } from "react-icons/ai";
import { Link, useNavigate, useParams } from "react-router-dom";

export default function SearchHeader() {
  const [text, setText] = useState<string>("");
  const navigate = useNavigate();
  const { keyword } = useParams();

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setText(e.target.value);
  };
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    navigate(`/videos/${text}`);
  };

  useEffect(() => setText(keyword || ""), [keyword]);

  return (
    <header className="w-full flex p-4 text-2xl border-b border-zinc-600">
      <Link to="/" className="flex items-center">
        <AiFillYoutube className=" text-4xl text-brand" />
        <h1 className="ml-1 font-bold text-3xl">Youtube</h1>
      </Link>
      <form className="w-full flex justify-center" onSubmit={handleSubmit}>
        <input
          className="w-7/12 p-2 bg-black text-gray-50 outline-none"
          type="text"
          placeholder="Search..."
          value={text}
          onChange={handleChange}
        />
        <button className="bg-zinc-600 px-4">
          <AiOutlineSearch />
        </button>
      </form>
    </header>
  );
}
