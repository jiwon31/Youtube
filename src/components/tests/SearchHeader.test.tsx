import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SearchHeader from "components/SearchHeader";
import { Route } from "react-router-dom";
import renderer from "react-test-renderer";
import { withRouter } from "tests/utils";

describe("SearchHeader", () => {
  it("renders", () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<SearchHeader />} />)
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders with keyword correctly", () => {
    render(
      withRouter(
        <Route path="/:keyword" element={<SearchHeader />} />,
        "/apple"
      )
    );
    expect(screen.getByDisplayValue("apple")).toBeInTheDocument();
  });

  it("navigate to results page on search button clicked", () => {
    const keyword = "fake-keyword";

    render(
      withRouter(
        <>
          <Route path="/home" element={<SearchHeader />} />
          <Route
            path={`/videos/${keyword}`}
            element={<p>{`Search result for ${keyword}`}</p>}
          />
        </>,
        "/home"
      )
    );

    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button");

    userEvent.type(input, keyword);
    userEvent.click(button);

    expect(
      screen.getByText(`Search result for ${keyword}`)
    ).toBeInTheDocument();
  });
});
