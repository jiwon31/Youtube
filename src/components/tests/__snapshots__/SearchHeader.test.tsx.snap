// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader renders 1`] = `
<header
  className="w-full flex p-4 text-2xl border-b border-zinc-600"
>
  <a
    className="flex items-center"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="text-4xl text-brand"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M941.3 296.1a112.3 112.3 0 0 0-79.2-79.3C792.2 198 512 198 512 198s-280.2 0-350.1 18.7A112.12 112.12 0 0 0 82.7 296C64 366 64 512 64 512s0 146 18.7 215.9c10.3 38.6 40.7 69 79.2 79.3C231.8 826 512 826 512 826s280.2 0 350.1-18.8c38.6-10.3 68.9-40.7 79.2-79.3C960 658 960 512 960 512s0-146-18.7-215.9zM423 646V378l232 133-232 135z"
      />
    </svg>
    <h1
      className="ml-1 font-bold text-3xl"
    >
      Youtube
    </h1>
  </a>
  <form
    className="w-full flex justify-center"
    onSubmit={[Function]}
  >
    <input
      className="w-7/12 p-2 bg-black text-gray-50 outline-none"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <button
      className="bg-zinc-600 px-4"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
        />
      </svg>
    </button>
  </form>
</header>
`;
